tests/project/test_project_schemas.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+from pydantic import ValidationError
+from snowflake.cli.api.project.schemas.project_definition import ProjectDefinition
+from snowflake.cli.api.project.schemas.snowpark.argument import Argument
+from snowflake.cli.api.project.schemas.snowpark.callable import FunctionSchema
+
+# TODO:
+# -test nice error message when update fails - this should be done by yaml loader, here we only check that error is raised
+# -test if runtime is passed correctly to Callable
+
+
+def test_if_fields_are_updated_correctly_by_assignment(argument_instance: Argument):
+    argument_instance.name = "Baz"
+
+    assert argument_instance.name == "Baz"
+
+
+def test_if_optional_fields_set_to_none_are_correctly_updated_by_assignment(
+    argument_instance: Argument,
+):
+    assert argument_instance.default == None
+
+    argument_instance.default = "Baz"
+
+    assert argument_instance.default == "Baz"
```

Comment:
Do these 2 tests test same thing?

Reply:
In the first one, the tested field is of type `str`, in second- it's `Optional[str]`- or `str | None`. And the field is initially set to none. 
It could be done in one test, but i decided to go with one responsibility per test principle.

Reply:
Ok, my bad. I thought you use same field in both tests.